jib-cli/src/main/java/com/google/cloud/tools/jib/cli/War.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli;
+
+import com.google.cloud.tools.jib.api.Containerizer;
+import com.google.cloud.tools.jib.api.JibContainer;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.LogEvent;
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.cli.logging.CliLogger;
+import com.google.cloud.tools.jib.cli.war.WarFiles;
+import com.google.cloud.tools.jib.plugins.common.globalconfig.GlobalConfig;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
+import com.google.cloud.tools.jib.plugins.common.logging.SingleThreadedExecutor;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Multimaps;
+import com.google.common.util.concurrent.Futures;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+    name = "war",
+    mixinStandardHelpOptions = true,
+    showAtFileInUsageHelp = true,
+    description = "Containerize a war")
+public class War implements Callable<Integer> {
+  @CommandLine.Spec
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.Mixin
+  @VisibleForTesting
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  CommonCliOptions commonCliOptions;
+
+  @CommandLine.Mixin
+  @VisibleForTesting
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  CommonContainerConfigCliOptions commonContainerConfigCliOptions;
+
+  @CommandLine.Parameters(description = "The path to the war file (ex: path/to/my-war.war)")
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private Path warFile;
+
+  @CommandLine.Option(
+      names = "--app-root",
```

Comment:
I think this option should be common to all the commands (`build`, `jar`, and `war`).

Reply:
Yes that is true. This is indeed a general purpose flag but I would also say that it is an absolute necessity for the `war` command. Since it is more of an enhancement for the other commands and this PR is already pretty large, I was thinking of doing this in a different PR. Although, I'm not sure how much of a difference adding `--app-root` would make to the `build` command since the creator of the buildfile can use the `dest` variable to copy files to a specific destination on the container?

Reply:
Oh, good point. The option doesn't make sense to the `build` command but only for `jar` and `war`. We don't have to do this in this PR.